tests/integration/kubernetes/gha-run.sh

```diff
@@ -240,14 +241,91 @@ function cleanup() {
 }
 
 function deploy_snapshotter() {
-	echo "::group::Deploying ${SNAPSHOTTER:-}"
-	#TODO Add the deployment logic for the snapshotter in PR https://github.com/kata-containers/kata-containers/pull/8585.
+	echo "::group::Deploying ${SNAPSHOTTER}"
+	case ${SNAPSHOTTER} in
+		nydus) deploy_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac	
 	echo "::endgroup::"
 }
 
 function cleanup_snapshotter() {
-	echo "::group::Cleanuping ${SNAPSHOTTER:-}"
-	#TODO Add the logic for cleaning up the snapshotter in PR https://github.com/kata-containers/kata-containers/pull/8585.
+	echo "::group::Cleanuping ${SNAPSHOTTER}"
+	case ${SNAPSHOTTER} in
+		nydus) cleanup_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac
+	echo "::endgroup::"
+}
+
+function deploy_nydus_snapshotter() {
+    echo "::group::deploy_nydus_snapshotter"
+	ensure_yq
+
+	local nydus_snapshotter_install_dir="/tmp/nydus-snapshotter"
+	if [ -d "${nydus_snapshotter_install_dir}" ]; then
+		rm -rf "${nydus_snapshotter_install_dir}"
+	fi
+	mkdir -p "${nydus_snapshotter_install_dir}"
+	nydus_snapshotter_url=$(get_from_kata_deps "externals.nydus-snapshotter.url")
+	nydus_snapshotter_version=$(get_from_kata_deps "externals.nydus-snapshotter.version")
+	git clone -b "${nydus_snapshotter_version}" "${nydus_snapshotter_url}" "${nydus_snapshotter_install_dir}"
+
+	pushd "$nydus_snapshotter_install_dir"
+	if [ "${PULL_TYPE}" == "guest-pull" ]; then
```

Comment:
Not a review comment, just to remind that PULL_TYPE is set to guest-pull in the pre-PR to this that updates the workflows: https://github.com/kata-containers/kata-containers/commit/a214bd8d1361699f8f7f77b5065d5e0282034525